plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/UpdateQueryGroupRequest.java

```diff
@@ -0,0 +1,152 @@
+        String name,
+        ResiliencyMode resiliencyMode,
+        Map<ResourceType, Double> resourceLimits,
+        long updatedAtInMillis
+    ) {
+        this.name = name;
+        this.resiliencyMode = resiliencyMode;
+        this.resourceLimits = resourceLimits;
+        this.updatedAtInMillis = updatedAtInMillis;
+    }
+
+    /**
+     * Constructor for UpdateQueryGroupRequest
+     * @param in - A {@link StreamInput} object
+     */
+    public UpdateQueryGroupRequest(StreamInput in) throws IOException {
+        super(in);
+        name = in.readString();
+        if (in.readBoolean()) {
+            resourceLimits = in.readMap((i) -> ResourceType.fromName(i.readString()), StreamInput::readDouble);
+        } else {
+            resourceLimits = new HashMap<>();
+        }
+        if (in.readBoolean()) {
+            resiliencyMode = ResiliencyMode.fromName(in.readString());
+        } else {
+            resiliencyMode = null;
+        }
+        updatedAtInMillis = in.readLong();
+    }
+
+    /**
+     * Generate a UpdateQueryGroupRequest from XContent
+     * @param parser - A {@link XContentParser} object
+     * @param name - name of the QueryGroup to be updated
+     */
+    public static UpdateQueryGroupRequest fromXContent(XContentParser parser, String name) throws IOException {
+        QueryGroup.Builder builder = QueryGroup.Builder.fromXContent(parser);
+        return new UpdateQueryGroupRequest(name, builder.getResiliencyMode(), builder.getResourceLimits(), Instant.now().getMillis());
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        QueryGroup.validateName(name);
+        if (resourceLimits != null) {
+            QueryGroup.validateResourceLimits(resourceLimits);
+        }
+        assert QueryGroup.isValid(updatedAtInMillis);
+        return null;
+    }
+
+    /**
+     * name getter
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * ResourceLimits getter
+     */
+    public Map<ResourceType, Double> getResourceLimits() {
+        return resourceLimits;
+    }
+
+    /**
+     * resiliencyMode getter
+     */
+    public ResiliencyMode getResiliencyMode() {
+        return resiliencyMode;
+    }
+
+    /**
+     * updatedAtInMillis getter
+     */
+    public long getUpdatedAtInMillis() {
+        return updatedAtInMillis;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(name);
+        if (resourceLimits == null || resourceLimits.isEmpty()) {
+            out.writeBoolean(false);
+        } else {
+            out.writeBoolean(true);
+            out.writeMap(resourceLimits, ResourceType::writeTo, StreamOutput::writeDouble);
+        }
+        if (resiliencyMode == null) {
+            out.writeBoolean(false);
+        } else {
+            out.writeBoolean(true);
+            out.writeString(resiliencyMode.getName());
+        }
```

Comment:
we can replace this with `out.writeOptionalString(resiliencyMode == null ? null : resiliencyMode.getName())`